File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/webrtcclient/Activities/PermissionDeniedActivity.java
package kore.botssdk.audiocodes.webrtcclient.Activities;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import kore.botssdk.R;
import kore.botssdk.audiocodes.webrtcclient.General.ACManager;
import kore.botssdk.audiocodes.webrtcclient.General.Prefs;
import kore.botssdk.audiocodes.webrtcclient.Permissions.PermissionManager;


public class PermissionDeniedActivity extends BaseAppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //this.requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.permission_denied_activity);

        initGui();
    }

    private void initGui() {

        Button continueButton = (Button)findViewById(R.id.permission_denied_button_continue);
        Button editButton = (Button)findViewById(R.id.permission_denied_button_edit);

        continueButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                boolean isFirstLogin = Prefs.isFirstLogin();
                if (isFirstLogin)
                {
                    startNextActivity(LoginActivity.class);
                    finish();
                }
                else {
                    //start login and open app main screen
                    ACManager.getInstance().startLogin();
                    startNextActivity(MainActivity.class);
                    finish();
                }
            }
        });

        editButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                PermissionManager.askForSystemAlertWindowPermission();
                finish();
            }
        });
    }

}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/webrtcclient/activities/PermissionDeniedActivity.kt
package com.kore.ui.audiocodes.webrtcclient.activities

import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.kore.ui.R
import com.kore.ui.audiocodes.webrtcclient.general.ACManager
import com.kore.ui.audiocodes.webrtcclient.general.Prefs
import com.kore.ui.audiocodes.webrtcclient.permissions.PermissionManager
import com.kore.ui.databinding.PermissionDeniedActivityBinding

class PermissionDeniedActivity : BaseAppCompatActivity() {
    private lateinit var binding: PermissionDeniedActivityBinding
    private val acManager: ACManager = ACManager.getInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.inflate(layoutInflater, R.layout.permission_denied_activity, null, false)
        setContentView(binding.root)
        initGui()
    }

    private fun initGui() {
        binding.permissionDeniedButtonContinue.setOnClickListener {
            if (Prefs.getIsFirstLogin(this)) {
                startNextActivity(LoginActivity::class.java)
                finish()
            } else {
                //start login and open app main screen
                acManager.startLogin(this)
                startNextActivity(MainActivity::class.java)
                finish()
            }
        }
        binding.permissionDeniedButtonEdit.setOnClickListener {
            PermissionManager.askForSystemAlertWindowPermission(this)
            finish()
        }
    }
}